### PR TITLE
Use the cookie_path config to set the messages cookie

### PR DESCRIPTION
**Pull request by joris-ati4**
### Description (*)
This is a fix to take into account the cookie path that has been configured in the magento backend when setting the mage-messages cookie.
If this is not set, there is an inconsistency between the PHP and JS where the lib/web/mage/cookies.js takes into account the path and the backend PHP code doesn't.

### Manual testing scenarios (*)

1. Set a cookie path at the store level in the Magento 2 configuration.
2. Go to the registration page and register an account in order to have a message displayed on the screen.
3. Open chrome dev tools and go to the cookies.
4. You will see 2 cookies registered, an empty cookie with the cookie path from the configuration and one cookie with a path = '/'
5. The JS puts the cookie with the configured cookie path while the php backend has the path hardcoded at /

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by thomas-kl1** [comment]
@magento run all tests

**Comment by jhart-efaktory** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Hi, Could you please cover this change by automatic test?  Thanks !

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by Andreeaisp** [comment]
@magento give me test instance

**Comment by Andreeaisp** [comment]
@magento give me 2.4-develop instance

File: app/code/Magento/Theme/view/frontend/web/js/view/messages.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ define([
         initialize: function () {
             this._super();
 
-            this.cookieMessages = _.unique($.cookieStorage.get('mage-messages'), 'text');
+            this.cookieMessages = _.unique($.mage.cookies.get('mage-messages'), 'text');
             this.messages = customerData.get('messages').extend({
                 disposableCustomerData: 'messages'
```
Comment (mrtuvn) [inline app/code/Magento/Theme/view/frontend/web/js/view/messages.js:34]: any reason why you change from cookieStorage to mage.cookies

**Comment by Den4ik** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by Den4ik** [comment]
@joris-ati4 Could you merge main line into your branch?

File: app/code/Magento/Theme/view/frontend/web/js/view/messages.js (hunk 1 above)
Comment (clivewalkden) [reply app/code/Magento/Theme/view/frontend/web/js/view/messages.js:34]: Don't think this is required, in my testing return type of $.mage.cookies.get('mage-messages') is a string which then breaks the underscore .unique() function and returns "["

**Comment by engcom-Hotel** [comment]
Hello @joris-ati4,

Thanks for the contribution!

After seeing conflicts, it seems that the issue is already been resolved and merged with the mainline. Hence we are closing this PR.

Let us know in case we have missed anything.

Thanks

**Pull request closed without merging**